src/olympia/amo/views.py

```diff
@@ -78,7 +79,10 @@ def handler403(request):
 
 @non_atomic_requests
 def handler404(request):
-    if request.path_info.startswith('/api/'):
+    if request.path_info.startswith('/api/v3/'):
```

Comment:
If we ever bump the API version I bet we'll forget about this. Could you not inspect the `Accept` request header instead to see if it's `json`?